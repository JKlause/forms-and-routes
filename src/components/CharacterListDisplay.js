import React from 'react';
// import styles from './Footer.css';
import PropTypes from 'prop-types';
import CharacterListItem from './CharacterListItem';

const CharacterListDisplay = ({ characters, handleCharacterClick }) => {
  
  const characterElements = characters.map((character, index) => {
    <li key={`${index}-${character.photoUrl}`}>
      <CharacterListItem handleCharacterClick={handleCharacterClick} {...character}/>
    </li>
  });
  
  return (
    <div>
      <h1>Characters</h1>
      <ul>
        {characterElements}
      </ul>
    </div>
  );
};

CharacterListDisplay.propTypes = {
  characters: PropTypes.array.isRequired,
  handleCharacterClick: PropTypes.func.isRequired
};


export default CharacterListDisplay;
