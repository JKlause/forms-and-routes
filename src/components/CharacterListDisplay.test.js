import React from 'react';
import { shallow } from 'enzyme';
import CharacterListDisplay from './CharacterListDisplay';

describe('CharacterListDisplay component', () => {
  const props = {
    characters: [],
    handleCharacterClick: () => {}
  };
  it('renders CharacterListDisplay', () => {   
    const wrapper = shallow(<CharacterListDisplay {...props}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
