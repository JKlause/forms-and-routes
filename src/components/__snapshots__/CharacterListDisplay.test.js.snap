// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterListDisplay component renders CharacterListDisplay 1`] = `
<div>
  <h1>
    Characters
  </h1>
  <ul />
</div>
`;
